opendssdirect/utils.py

```diff
@@ -279,3 +279,44 @@ def xycurves_to_dataframe(dss=None):
     if dss is None:
         import opendssdirect as dss
     return to_dataframe(dss.XYCurves)
+
+
+def iter_elements(element_class, element_func):
+    """Yield the return of element_func for each element of type element_class.
+
+    Parameters
+    ----------
+    element_class : class
+        Subclass of opendssdirect.CktElement
+    element_func : function
+        Function to run on each element
+
+    Yields
+    ------
+    Return of element_func
+
+    Examples
+    --------
+    >>> import opendssdirect as dss
+    >>> def get_reg_control():
+        return {
+            "name": dss.RegControls.Name(),
+            "enabled": dss.CktElement.Enabled(),
+            "transformer": dss.RegControls.Transformer(),
+        }
+
+    # Iterate over each RegControl.
+    >>> for reg_control in iter_elements(dss.RegControls, get_reg_control):
+        print(reg_control["name"])
+
+    >>> def get_reg_controls():
+        return list(iter_elements(dss.RegControls, get_reg_control))
+
+    # Create a list of RegControl dictionaries.
+    >>> reg_controls = get_reg_controls()
+        
+    """
+    element_class.First()
```

Comment:
Is there a way to validate whether this will work before we call it?  For example, you can't do this for Bus.

Reply:
Not at the moment, but with the impending rewrite we should be able to do it.